WIP: Wayland proxy and PipeWire support

lxp opened the pull request.

lxp committed:
WIP: Add PipeWire support

lxp committed:
TEMP: Add debug messages

freebritney132 commented:
Is there a reason why you would want/need the wayland socket to be sandboxed? Same for pipewire

lxp commented:
> Is there a reason why you would want/need the wayland socket to be sandboxed? Same for pipewire

Wayland:
* Prefix window titles with the sandbox name (i.e. to have titles like `[firefox] Mozilla Firefox`....), so that the different sandboxes can be distinguished and they cannot spoof windows you would expect in another sandbox.
* Make sure dangerous Wayland extensions are not available.
* Both could be handled by the Wayland compositor, if it supports security contexts.
* Ideally, there would also be a more secure solution for clipboard management. This is still an unsolved issue with `wayland-proxy-virtwl`. I am open for ideas.

Pipewire:
* If Pipewire is not sandboxed, it is possible to playback and capture on all audio/video devices. Furthermore, it is possible to listen in and modify playback of all other applications.
* I am not using a separate proxy application here, but just a functionality of Pipewire (`pw-container`) to allow it to distinguish between sandboxes. It is a bit similar to what Flatpak and Snap already does. More concretely, it tries to emulate what Snap does. However, in the end I also rewrote the Pipewire access check script, since I felt the original one was too lax for my taste.

freebritney132 commented:
That's all about what I gathered, thanks for the thorough answer.

lxp committed:
TEMP: Experiment with executable paths outside /bin

max-privatevoid committed:
Merge pull request #93 from lxp/pasta-support

Add support for user-space networking via pasta

max-privatevoid committed:
add a basic devShell

max-privatevoid committed:
launcher: fix deprecations

max-privatevoid committed:
launcher: go mod tidy

max-privatevoid committed:
modules/gpu: mesa.drivers -> mesa

max-privatevoid committed:
contrib: expose modules and presets via nixpakModules

max-privatevoid committed:
contrib: include modules from nixpak/pkgs

max-privatevoid committed:
contrib: include package presets from nixpak/pkgs

max-privatevoid committed:
contrib: expose packages

max-privatevoid committed:
contrib: include packages from nixpak/pkgs

max-privatevoid committed:
remove i686-linux support

max-privatevoid committed:
contrib: docs

max-privatevoid committed:
Merge pull request #152 from nixpak/merge-pkgs

Merge nixpak/pkgs

max-privatevoid committed:
support flatpak instance metadata

Co-authored-by: Jan B. <git@keksgesicht.de>

max-privatevoid committed:
Merge pull request #155 from nixpak/flatpak-instances

Support flatpak instance metadata

max-privatevoid committed:
contrib: actually include preset in packages

lxp committed:
Passthrough app config

This allows accessing the original Nixpak app config from the
Nixpak derived package. For example, to access the wrapped package.

lxp committed:
Add Wayland proxy support

This adds support for proxing the Wayland socket via
wayland-proxy-virtwl. It limits the supported Wayland protocols
to a sane set and allows prefixing window titles.
Although wayland-proxy-virtwl currently does not fully validate the
requests relayed to the Wayland compositor, it is already useful.
Furthermore, it looks like it might get used in future Qubes OS
releases for safeguarding the compositor. See:
https://github.com/talex5/wayland-proxy-virtwl/issues/90
https://github.com/talex5/ocaml-wayland/pull/49

lxp committed:
Add option to unset all environment variables
